### ISSUE TITLE
Continuing maintenance?

### ISSUE THREAD

**Issue by sundeepgupta** [kiwi-bdd/Kiwi#662]
Just wondering plans for maintenance on this framework going forward. Thanks.

**Comment by programmerdave** [comment]
+1

**Comment by ecaselles** [comment]
+1

**Comment by samjarman** [comment]
+1

**Comment by programmerdave** [comment]
#306

**Comment by programmerdave** [comment]
Similar to #306

**Comment by lolgear** [comment]
@programmerdave this repository has million open issues, why not to close most of them? ( It would be cleaner to focus on really needed issues). This one should be closed, no?

**Comment by sharplet** [comment]
Hey everyone. Big apologies for the silence lately—it's super lame. :neutral_face: 

I just want to put this out there: I'd love to add some new collaborators. I'm not aware of a documented policy, but I'm open to talking about it and I think this project would benefit from some much-needed TLC.

If you're interested in become a maintainer or contributing in some other way, please post here.

Pinging @kiwi-bdd/core-team, please speak up if you have any input on this!

**Comment by lolgear** [comment]
@sharplet I would like to contribute and focus on my strange issue ( dictionaries are not equal ) with another annoying issue ( Spec always pass when running alone )

**Comment by trimmurrti** [comment]
@sharplet I would like to contribute as well. Could you please specify, what are the plans for 3.0 with a bit more explanations, perhaps? Because the link you gave is a bit opaque in my opinion, so I'm not really sure, if I would have enough time to handle it.

Also, could you please tell, if Kiwi is still supported, as I created a pull-request some time ago, but didn't receive any answer up till now.

**Comment by modocache** [comment]
Hey all! I spoke with @sharplet and @orta, and we've decided we're going to move Kiwi to v3.0, which will be Kiwi's last major version. Kiwi will be "done" at that point: a finished product. We'll happily accept pull requests for bug fixes, and release minor version updates (like v3.0.1) in order to push those fixes out to end users. If technologies revealed at WWDC causes minor issues, we'll fix those too.

However, we probably won't merge pull requests for new features. For example, features like shared examples (#442, #558, #681) or custom output formatters (#449) are outside of the scope of this project. We're focusing our time on other projects and other aspects of our lives -- we won't be spending time building, reviewing, maintaining, or improving large new features like these.

Of course, you may continue to use Kiwi. If you encounter bugs, please report them and, if possible, contribute fixes. If you'd like to build entirely new features for Kiwi, we recommend you fork the project and continue building there.

We'll make a more "official" announcement once we actually tag a version as v3.0, but I thought I'd post this now to let you all know what's up. Thanks for using and supporting Kiwi over the years. I know I had a blast working on it since my first commit back in 2012. 🙇

**Comment by lolgear** [comment]
@modocache please, don't freeze shared examples! They are so awesome and can not be missed from 3.0 release :(  
Besides they add necessary feature ( regression testing for/after refactoring ).

**Comment by modocache** [comment]
@lolgear I hear you. Believe me, it's a tough decision. Still, Kiwi's current set of maintainers just don't have the bandwidth to develop an entirely new feature and make sure it's stable before releasing a finished product. 3.0 isn't intended to be a feature release, after all -- it's intended to mark a finishing point.

**Comment by ecaselles** [comment]
@modocache Thanks for the update!

I totally respect your decision and it feels good to do a final release and freeze new features, leaving Kiwi as an stable finished product 👍 

However, correct me if I am wrong, but **shared examples are not a new feature**. They have been part of Kiwi since [v2.4.0](https://github.com/kiwi-bdd/Kiwi/releases/tag/v2.4.0), after #600 was merged. Does this mean that v3.0 will not include new changes on shared examples but still include them as in their current implementation? Or that you will just remove them entirely and make v3.0 come from a previous version? 😕

**Comment by modocache** [comment]
Uhhmm... sorry, not only did I forget that Kiwi implements shared examples, I forgot that **I reviewed your pull request to add them**!! 😱 💦 

My bad!! Sorry that was so confusing. My point was that we wouldn't be working on any *new* features for v3.0. As you pointed out, shared examples already exist, and we won't be going out of our way to remove them.

Sorry again!! Phew, that's probably the most embarrassing thing I'll do on the Internet for at least... well, a few hours. 😅

**Comment by trimmurrti** [comment]
@modocache Is that the final decision? Wouldn't it be a better idea to transition maintainer rights to some other dev in order to continue the development? Most of my project testing routines are based on Kiwi and it is really sad, that the maintenance and the development is going to be terminated.

**Comment by orta** [comment]
I think the point here, is the development is stopping from the current maintainers, not the maintenance.

[Specta/Expecta](https://github.com/specta/specta#status) has been this way for about a year, and it's not going to be a problem for people to continue relying on those projects either. I still do, and will for the foreseeable future. The point of @modocache's message above was _you shouldn't be worried about existing projects breaking with Kiwi_.

> We'll happily accept pull requests for bug fixes, and release minor version updates (like v3.0.1) in order to push those fixes out to end users. If technologies revealed at WWDC causes minor issues, we'll fix those too.

Looking at the [contributions graph](https://github.com/kiwi-bdd/Kiwi/graphs/contributors) the project has been pretty inactive for two years, and before last month there were a very large amount of 0 response issues/PRs. Fresh blood in maintainers could solve this, but Obj-C testing frameworks are solid, mature platforms to work with. People are moving to Swift, and not taking Kiwi with them. That's OK.

**Comment by trimmurrti** [comment]
The thing here is, that there are loads of features, that could be implemented. That was the main idea behind my message. That the maintenance will only be till v3.0.1 and WWDC. Afterwards it will be stalled even in bugfixing department, not even to mention new features. Perhaps, I did misunderstood it, but that's what I got from the @modocache message.

Do you mean, that the framework is left alone, because Swift is being more popular nowadays?

**Comment by ecaselles** [comment]
@modocache No problem at all 😄 . It was a bit confusing at the beginning, but it all makes sense now. 👏

**Comment by ecaselles** [comment]
@sharplet @modocache @orta how is v3.0 coming along? 🤔 I am aware that no new features will be added. But I think we have got some good maintenance work (e.g. [removing annoying warnings](https://github.com/kiwi-bdd/Kiwi/pull/656)) that have been merged into master since the last release ([2.4.0](https://github.com/kiwi-bdd/Kiwi/commit/974f95851e290cd588b460a12eecd9a42c21bdc6)) and it would be great to be able to get those changes in our projects without having to specifically point the pod to master. Happy to help if needed.

**Comment by sharplet** [comment]
@ecaselles Thanks for your patience so far. I've just sent you an invite to the team, so if you can help move things along that would be fantastic!

**Comment by ecaselles** [comment]
Thanks @sharplet. Let's make it happen 💪
